src/scheduler.rs

```diff
@@ -130,6 +130,23 @@ async fn inbox_loop(ctx: Context, started: Sender<()>, inbox_handlers: ImapConne
                         }
                     };
 
+                    match ctx.get_config_bool(Config::FetchedExistingMsgs).await {
+                        Ok(fetched_existing_msgs) => {
+                            if !fetched_existing_msgs {
+                                if let Err(err) = connection.fetch_existing_msgs(&ctx).await {
+                                    warn!(ctx, "Failed to fetch existing messages: {:#}", err);
+                                }
+                            }
+                        }
+                        Err(err) => {
+                            warn!(
+                                ctx,
+                                "Failed to check if we already fetched existing messages: {:#}",
+                                err
+                            );
```

Comment:
Nit:
```suggestion
                            warn!(ctx, "Can't get Config::FetchedExistingMsgs: {:#}", err);
```

Easier to read IMHO, and long explaining error messages aren't important here since if it fails, it's usually database errors, anyway (in fact, `warn!("{:#}", err)` would prob. be OK, too)